website/docs/r/record.html.markdown

```diff
@@ -85,19 +85,34 @@ resource "ns1_record" "www" {
     }
   }
 }
+
+# Some other non-NS1 provider that returns a zone with a trailing dot.
+resource "external_source" "baz" {
+  zone = "terraform.example.io."
+}
+
+# Basic record showing how to clean a zone or domain field that comes from
+# another non-NS1 resource. DNS names often end in '.' characters to signify
+# the root of the DNS tree, but the NS1 API does not support this.
+#
+# In other cases, a domain or zone may be passed in with a preceding dot ('.')
+# character which would likewise lead the system to fail.
+resource "ns1_record" "external" {
+  zone   = replace("${external_source.zone}", "/(^\\.)|(\\.$)/", "")
+  domain = replace("www.${external_source.zone}", "/(^\\.)|(\\.$)/", "")
```

Comment:
also pedantic, but we could make the domain example have a leading dot, maybe another var from the "baz" resource.